File: 2249-count-the-hidden-sequences/count-the-hidden-sequences.go
func numberOfArrays(differences []int, lower int, upper int) int {
    var start, maxVal, minVal int64 = 0, 0, 0;

    for _, diff := range differences {
        start += int64(diff);

        if start > maxVal {
            maxVal = start;
        }

        if start < minVal {
            minVal = start;
        }
    }

    rangeDiff := int64(upper - lower);
    valid := rangeDiff - (maxVal - minVal) + 1;

    if valid < 0 {
        return 0;
    }

    return int(valid);
}

File: 2249-count-the-hidden-sequences/count-the-hidden-sequences.java
class Solution {
    public int numberOfArrays(int[] differences, int lower, int upper) {
        long prefix = 0;
        long minPrefix = 0, maxPrefix = 0;
        
        for (int diff : differences) {
            prefix += diff;
            minPrefix = Math.min(minPrefix, prefix);
            maxPrefix = Math.max(maxPrefix, prefix);
        }
        
        long minStart = lower - minPrefix;
        long maxStart = upper - maxPrefix;
        
        long result = maxStart - minStart + 1;

        return (int)Math.max(result, 0);
    }
}